upup/pkg/fi/nodeup/command.go

```diff
@@ -341,6 +377,57 @@ func evaluateSpec(c *api.Cluster) error {
 	return nil
 }
 
+func evaluateSecurityGroups(vpcId string) ([]string, error) {
```

Comment:
This approach is tricky - I don't know if this will work with e.g. terraform.  That said we can leave it for now.

I think the Lyft CNI provider would need to change so that it discovers the security groups using tags or similar.

I guess because it only affects the lyft cni provider we could still merge this, though.

Reply:
Its an interesting thought. I will look into proposing some default auto-detection of security groups. I'm not sure that this would necessarily affect terraform, the instances will come up and the detection of the security groups would happen as the node comes up. I'm going off the assumption that the terraform would create basically the same ASGs that have the same cloud init logic, I haven't used it personally.